packages/client/src/__tests__/integration/happy/disconnect-finally/test.ts

```diff
@@ -24,4 +24,6 @@ test('disconnect-finally', async () => {
 
   const data = await run()
   expect(data).toMatchSnapshot()
+
+  await prisma.$disconnect()
```

Comment:
This tests was blocking the CI/jest because doing `await` on the prisma promise re-wakes the engine, even afyter a disconnect - which is the purpose of the test. However, the test did not do a final `$disconnect` after doing that.

Reply:
Why did this never cause a problem before?
Or is this one of the flaky tests we had for ages?

Reply:
By calling `disconnect` without await, we were basically pushing it down the event loop. For some reason, that has worked to happen after `await res`. So maybe the refactor has changed the order of these async events, I'm not certain.

Reply:
I did a check on the `main` branch:
```ts
  const r = prisma.user.findMany();

  await r
  prisma.$disconnect()
  await r
```
:arrow_up:  Does send the query twice and exits properly (with `binary` of course). While:
```ts
  const r = prisma.user.findMany();

  prisma.$disconnect()
  await r
  await r
```
:arrow_up: Will prevent the program from exiting (still with `binary`). Confirms my theory.